ucm2/Intel/sof-hda-dsp/HiFi-analog-sof.conf

```diff
@@ -0,0 +1,492 @@
+	      HeadphoneFirBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/eq_fir/pass.blob"
+	      HeadphoneDrcBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/drc/passthrough.blob"
+	}
+}
+
+# Include blob customizations: sys_vendor / product_name .conf
+# TODO: The 2nd level could be instead product_sku .conf ?
+Define.ProductConfigSOF "/sof/product_configs/${sys:devices/virtual/dmi/id/sys_vendor}/${sys:devices/virtual/dmi/id/product_name}.conf"
+Define.SOFConfFullPath "${ConfTopDir}${var:ProductConfigSOF}"
+
+If.HasProductConfig {
+	Condition {
+		Type Path
+		Mode read
+		Path "${var:SOFConfFullPath}"
+	}
+	True {
+		Include.product-config.File "${var:ProductConfigSOF}"
+	}
+}
+
+If.imicname {
+	Condition {
+		Type ControlExists
+		Control "name='Internal Mic Boost Volume'"
+	}
+	True.Define {
+		FrontMicName "Internal"
+	}
+}
+
+If.fmicname {
+	Condition {
+		Type ControlExists
+		Control "name='Front Mic Playback Switch'"
+	}
+	True.Define {
+		FrontMicName "Front"
+		FrontMicJack "Front Mic Jack"
+	}
+}
+
+If.fmic {
+	Condition {
+		Type String
+		Empty "${var:FrontMicName}"
+	}
+	False {
+		If.mic2 {
+			Condition {
+				Type String
+				String1 "${var:DeviceMic}"
+				String2 "Mic2"
+			}
+			True.Define {
+				DeviceFrontMic "Mic3"
+			}
+			False.Define {
+				DeviceMic "Mic1"
+				DeviceFrontMic "Mic2"
+			}
+		}
+	}
+}
+
+If.lineout {
+	Condition {
+		Type ControlExists
+		Control "name='Line Out Playback Volume'"
+	}
+	True.Define {
+		loctl "Line Out"
+		lovol "Line Out"
+	}
+}
+
+If.linefront {
+	Condition {
+		Type ControlExists
+		Control "name='Front Playback Volume'"
+	}
+	True.Define {
+		LineDevice "1"
+	}
+}
+
+If.hplo {
+	Condition {
+		Type ControlExists
+		Control "name='Headphone+LO Playback Volume'"
+	}
+	True.Define {
+		hpvol "Headphone+LO"
+		lovol "Headphone+LO"
+	}
+}
+
+If.hpjack {
+	Condition {
+		Type ControlExists
+		Control "iface=CARD,name='Headphone Mic Jack'"
+	}
+	True.Define.hpjack "Headphone Mic Jack"
+	False.If.front {
+		Condition {
+			Type ControlExists
+			Control "iface=CARD,name='Front Headphone Jack'"
+		}
+		True.Define.hpjack "Front Headphone Jack"
+	}
+}
+
+SectionDevice."Headphones" {
+	Comment "Headphones"
+
+	If.headphone_switch {
+		Condition {
+			Type ControlExists
+			Control "name='Headphone Playback Switch'"
+		}
+		True.If.headphone_with_drc {
+			Condition {
+				Type ControlExists
+				Control "name='${var:PostMixerAnalogPlaybackDRCSwitch}'"
+			}
+			True {
+				EnableSequence [
+					cset "name='Headphone Playback Switch' on"
+					cset "name='${var:PostMixerAnalogPlaybackDRCSwitch}' off"
+					cset-tlv "name='${var:PostMixerAnalogPlaybackDRCBytes}' ${var:HeadphoneDrcBlob}"
+					cset-tlv "name='${var:PostMixerAnalogPlaybackIIRBytes}' ${var:HeadphoneIirBlob}"
+					cset-tlv "name='${var:PostMixerAnalogPlaybackFIRBytes}' ${var:HeadphoneFirBlob}"
+					#shell "/bin/echo 'Headphone DRC EQ ${var:HeadphoneIirBlob} ${var:HeadphoneFirBlob} ${var:SOFConfFullPath}' >> /tmp/alsa-ucm.txt"
```

Comment:
is this really a sequence that needs to be done when enabling an endpoint, or could it be done at boot time?

Boot time sequences are stored separately in a different file.

Reply:
and now that I think of it, I wonder if the only difference between the processing-enabled cases or not enabled should be the boot settings.

In theory when enabling a path the controls are restored by DAPM, are they not? If I am correct we could simplify all this and just provide a bunch of settings. Maybe some of the values will not be used, e.g. if a user never enables a given path, but that's considerably less complicated to maintain.

Reply:
This is needed with HDA with shared DAI for endpoints. I'm planning to add for EQ components similar on/off switch as DRC has. With such it would be possible to have boot time passed blob and passthrough for anything else but speaker and just control the switch control.  But it would also limit customize of other endpoints with this framework. So, I chose to transmit the blob in every EnableSequence. But e.g. with SDW with separate PCMs and DAIs for endpoints there is no need to do runtime blobs passing. I should work with such topology next since SDW PC notebooks are appearing.

The EQ firmware also has capability for enum type EQ presets (but no control exposed) so such could work for convertible notebooks, if UCM would be aware of orientation. I'll implement first the switch. The enum control add needs some thinking it it's feasible.

Note: DRC would work in git main builds with just switch control, and I don't currently see need for playback DRC for other than notebook kind of speaker. But the signed FW release that I'm using has non-working switch control. It requires a stream start for switch to get and hear the boost impact. While with blob send it always works. So, in this work I'm also always sending the DRC blob. Hope the new FW releases include this DRC fix.

Reply:
Ah yes for HDaudio the settings depend on how the endpoint is configured (speaker or headphone). Same for input if there's no PCH-DMIC.

Indeed I don't see how we can avoid configurations during the Enable step. Thanks for the clarification.